### ISSUE TITLE
beamlattice minLength unclear definition

### ISSUE THREAD

**Issue by alexcarruesco** [3MFConsortium/spec_beamlattice#33]
Hi,
I would like to discuss the definition for the minLength attribute in the beamlattice element because I found an inconsistency. The spec says:
"A producer MUST specify the minimal length of all beams in the lattice. The producer MUST NOT produce zero length beams (i.e. shorter than minlength). The consumer MUST ignore all beams with length shorter than minlength."

The parenthesis clarification in the text says that we understand “zero length beams” as those that are “shorter than minLength”. The problem appears from the combination of these 2 sentences:

- "The producer MUST NOT produce zero length beams (i.e. shorter than minlength)". This sentence implies that being this a MUST for the producer, any consumer should reject a 3MF if this does not hold for some beam, otherwise the consumer would be accepting 3MF files that do not comply with this sentence from the spec.

- "The consumer MUST ignore all beams with length shorter than minlength.". This sentence looks to me as opposite to the previous one, because this one says that the consumer MUST ignore the beams shorter than minlength, but the previous sentence just said that a beam with length < minLength does not comply with the spec, and the consumer should reject 3MFs that are not compliant.

Then, what should the consumer do when a beam with length < minlength is found? Should the 3mf be rejected, or the beam just ignored?

Personally, I would advocate for the rejection of the 3MF because if that situation happens it indicates a bad design of the lattice, and if we just ignore that problem the consumer will 3D print physical objects that does to represent the intent of the user. In the worst scenario, if we silently skip the non-conformal beams we could be losing big portions of the lattice if all the related beams have a length < minlength.

To sum up, I would propose to just remove the sentence "The consumer MUST ignore all beams with length shorter than minlength." from the spec and reject the 3MF with invalid beams. What do you think?

Note: This is related to issue #64 in the test_suites repository: https://github.com/3MFConsortium/test_suites/issues/64

**Comment by jordig100** [comment]
Propose to change the wording to:

“A producer MUST specify the minimal length of all beams in the lattice. The producer SHOULD NOT produce zero length beams (i.e. shorter than minlength). The consumer MUST ignore all beams with length shorter than minlength in the local mesh coordinate system before applying any transform”.

This removes the inconsistency in previous definition. Second it clarifies that the minlength is specified in the local coordinate systems as specified in the vertices before being transformed to the build coordinate system.
